Update WPT for Shadow DOM and update declarative Shadow DOM implementation

rniwa opened the pull request:
#### 27b052173dc3e137f14e8ff46e4b4797963a24bf
<pre>
Update WPT for Shadow DOM and update declarative Shadow DOM implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=249211">https://bugs.webkit.org/show_bug.cgi?id=249211</a>

Reviewed by Chris Dumez.

Update the web platform tests for shadow DOM and WebKit&apos;s declarative Shadow DOM implementation.
Rename shadowroot content attribute to shadowrootmode and enable streaming by default.

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/crashtests/move-to-new-tree-1343016.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/crashtests/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-after-attachshadow.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-parser-interaction.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-attachment.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-basic.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-basic.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-opt-in.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-with-disabled-shadow.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/getinnerhtml.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/innerhtml-before-closing-tag.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/innerhtml-before-closing-tag.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/innerhtml-on-ordinary-template.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/move-template-before-closing-tag.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/script-access.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/script-access.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/support/declarative-child-frame.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/delegatesFocus-highlight-sibling-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/delegatesFocus-highlight-sibling.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-default-tabindex-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-default-tabindex.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-2levels-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-2levels.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-delegatesFocus-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-delegatesFocus.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-fallback-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-fallback.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-fallback-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-fallback.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-with-tabindex-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-with-tabindex.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-web-component-input-type-radio-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-web-component-input-type-radio.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-with-delegatesFocus-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-with-delegatesFocus.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-nested-slots-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-nested-slots.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassignable-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassignable-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassigned-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassigned-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-unassignable-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-unassignable-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-with-negative-index-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-with-negative-index.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/resources/focus-utils.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/resources/shadow-dom.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/resources/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-method-with-delegatesFocus-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-method-with-delegatesFocus.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-shadowhost-display-none-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-shadowhost-display-none.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-shadow-dom-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-sibling-different-slots-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-sibling-different-slots-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-sibling-different-slots.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/offsetTop-offsetLeft-across-shadow-boundaries-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/offsetTop-offsetLeft-across-shadow-boundaries.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/w3c-import.log:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot-expected.txt: Added.

* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
* Source/WebCore/html/HTMLAttributeNames.in:
* Source/WebCore/html/HTMLTemplateElement.cpp:
(WebCore::HTMLTemplateElement::attachAsDeclarativeShadowRootIfNeeded):
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::HTMLConstructionSite::insertHTMLTemplateElement):

Canonical link: <a href="https://commits.webkit.org/257820@main">https://commits.webkit.org/257820@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d3c7257036929fda887ce04236c0f65fd1b18352

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100083 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9249 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33158 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109425 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169687 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10133 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86752 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92522 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107316 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105851 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7659 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90940 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34367 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89565 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22334 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3031 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23849 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2995 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/46216 "Build is being retried. Recent messages:Cleaned up git repository; Skipping applying patch since patch_id isn't provided; Checked out pull request; Pull request contains relevant changes; Skipped layout-tests; layout-tests (retry)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9127 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43330 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4842 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2762 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-basic.tentative-expected.txt:
```diff
@@ -1,6 +1,7 @@
 
 PASS Declarative Shadow DOM: Basic test
 FAIL Declarative Shadow DOM: Feature detection assert_true: Unable to feature detect expected true got false
+FAIL Shadowrootmode reflection assert_equals: The shadowRootMode IDL should reflect the content attribute expected (string) "open" but got (undefined) undefined
```
These 2 failures here don't seem great.

rniwa replied:
Yeah, these are new test cases that are failing. Will post a patch to fix it as a followup.

cdumez reviewed (approved):
LGTM

rniwa committed:
Update WPT for Shadow DOM and update declarative Shadow DOM implementation
https://bugs.webkit.org/show_bug.cgi?id=249211

Reviewed by Chris Dumez.

Update the web platform tests for shadow DOM and WebKit's declarative Shadow DOM implementation.
Rename shadowroot content attribute to shadowrootmode and enable streaming by default.

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/crashtests/move-to-new-tree-1343016.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/crashtests/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-after-attachshadow.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-parser-interaction.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-attachment.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-basic.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-basic.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-shadow-dom-opt-in.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/declarative-with-disabled-shadow.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/getinnerhtml.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/innerhtml-before-closing-tag.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/innerhtml-before-closing-tag.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/innerhtml-on-ordinary-template.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/move-template-before-closing-tag.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/script-access.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/script-access.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/declarative/support/declarative-child-frame.html:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/delegatesFocus-highlight-sibling-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/delegatesFocus-highlight-sibling.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-default-tabindex-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-default-tabindex.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-fallback.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-2levels-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-2levels.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-delegatesFocus-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-delegatesFocus.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-fallback-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested-fallback.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-nested.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-fallback-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-fallback.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-shadow-in-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-with-tabindex-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slot-with-tabindex.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-web-component-input-type-radio-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-web-component-input-type-radio.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-with-delegatesFocus-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-with-delegatesFocus.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-nested-slots-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-nested-slots.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassignable-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassignable-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassigned-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-reverse-unassigned-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-unassignable-slot-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-unassignable-slot.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-with-negative-index-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-with-negative-index.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/resources/focus-utils.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/resources/shadow-dom.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/resources/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-method-with-delegatesFocus-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-method-with-delegatesFocus.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-shadowhost-display-none-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/focus-shadowhost-display-none.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/focus/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-shadow-dom-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-sibling-different-slots-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-sibling-different-slots-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/invalidate-sibling-different-slots.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/offsetTop-offsetLeft-across-shadow-boundaries-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/offsetTop-offsetLeft-across-shadow-boundaries.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/w3c-import.log:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/shadow-dom/focus-navigation/focus-navigation-slots-in-slot-expected.txt: Added.

* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
* Source/WebCore/html/HTMLAttributeNames.in:
* Source/WebCore/html/HTMLTemplateElement.cpp:
(WebCore::HTMLTemplateElement::attachAsDeclarativeShadowRootIfNeeded):
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::HTMLConstructionSite::insertHTMLTemplateElement):

Canonical link: https://commits.webkit.org/257820@main
